app/src/main/java/com/android/solvit/shared/ui/chat/ChatScreen.kt

```diff
@@ -474,7 +474,7 @@ fun AiSolverHeader(navigationActions: NavigationActions) {
         }
       },
       navigationIcon = {
-        IconButton(onClick = { navigationActions.navigateTo(Route.SERVICES) }) {
+        IconButton(onClick = { navigationActions.goBack() }) {
```

Comment:
✅ Good use of simple back navigation for chat ❗ However, there's no handling of empty back stack cases 🔧 Suggestion: Add a fallback destination:

```suggestion
IconButton(
    onClick = { 
        if (!navigationActions.canGoBack()) {
            navigationActions.navigateTo(Route.INBOX)
        } else {
            navigationActions.goBack()
        }
    }
) {
    Icon(Icons.Default.ArrowBack, contentDescription = "Back")
}
```